watt.py

```diff
@@ -227,18 +252,49 @@ class WDLTest:
     test_inputs: str
     expected_outputs: str or None
     cromwell_config: CromwellConfig
+    logger: argparse.FileType('w')
+
+    def get_log_path_str(self) -> str:
+        # If stem is dir, start filename without '-'
+        stem_sep = '-' if self.cromwell_config.log_prefix[-1] != '/' else ''
+        return f'{self.cromwell_config.log_prefix}{stem_sep}{self.workflow_name}-{self.test_name}.log'
 
     def run_test(self) -> TestResult:
         """
         Creates the return TestResult object by running Cromwell, and comparing the actual outputs to the expected ones.
         If no expected JSON is provided, then assumed the user expects the run to fail.
         """
+        self.print_startup()
+
         # Create Cromwell subprocess with self parameters
         output_path = f'cromwell-executions/watt/result-{self.workflow_name}-{self.test_name}-outputs.json'
-        stem_sep = '-' if self.cromwell_config.log_prefix[-1] != '/' else ''  # If stem is dir, start filename without '-'
-        log_path = f'{self.cromwell_config.log_prefix}{stem_sep}{self.workflow_name}-{self.test_name}.log'
-        cromwell_result = self.cromwell_config.run(self.path, self.test_inputs, output_path, log_path)
+        log_path_str = self.get_log_path_str()
+        cromwell_process = self.cromwell_config.process(self.path, self.test_inputs, output_path)
+
+        # parse cromwell_process stdout for workflow id, and also write to log_path
+        log_path = Path(log_path_str)
+        log_path.parent.mkdir(parents=True, exist_ok=True)
 
+        with open(log_path_str, 'wb') as cromwell_log_file:
+            while True:
+                line = cromwell_process.stdout.readline()
```

Comment:
Are you sure this can't get stuck in an infinite loop? How will it know to stop looking for new lines in stdout? And the benefit of doing it this way is to get both stdout and the logfile to hold the same logs?

Reply:
I don't think it should, get stuck in a infinite loop, because readline will eventually return null once the pipe is closed (I believe that is how it works).  I took this from https://stackoverflow.com/questions/2804543/read-subprocess-stdout-line-by-line.  Looking more closely at that stackoverflow question, though, it looks like the bug that requires this pattern was fixed in python 3, so can instead use the much more idiomatic 
```python
for line if cromwell_process.stdout:
```
 so I will switch to that.
 
 The point of going about it this way is to be able to parse the cromwell logs to look for the workflow id, and then print that into the watt logs (https://github.com/kachulis/watt/blob/bc152253327122e9a9a710f9e7d96a5f0a1f7cc0/watt.py#L285-L286).  This makes it much easier to track down the correction execution location in the event a test fails, without having to go into the individual cromwell log files.  You end up with the extra line like in this example:
 ```
 [AggregatePRSResults/base]   Starting test base for workflow AggregatePRSResults...
[AggregatePRSResults/base]     Workflow path: ./ImputationPipeline/AggregatePRSResults.wdl
[AggregatePRSResults/base]     Test inputs: ./test/AggregatePRSResults/test_inputs.json
[AggregatePRSResults/base]     Expected outputs: ./test/AggregatePRSResults/test_outputs.json
[AggregatePRSResults/base]     Cromwell log: cromwell_logs/AggregatePRSResults-base.log
[AggregatePRSResults/base]   Workflow id: 92f4d6cb-dc8a-4999-a3ed-2615317df287
``` 

also realizing that last line should be indented like the others, so will fix that.